t/enqueue.t

```diff
@@ -0,0 +1,186 @@
+
+    # Run a commandline invocation and check output against provided rx
+    sub enqueue {
+        my $cmd                = shift;
+        my $match_rx           = shift;
+        my $expect_match_count = shift;
+	my $verbose            = shift || 0;
+
+        my $full_cmd = "perl bin/enqueue.pl $cmd";
+	my (undef, undef, $line) = caller;
+        print "Check output from command on line $line [$full_cmd] against rx [$match_rx]\n" if $verbose;
+
+        # Run command, capture output.
+	# Use double spaces to make it look good.
+        my @cmd_output = `$full_cmd`;
+	my $double_space = "  ";
+        print "Output:[\n$double_space" . join($double_space, @cmd_output) . "]\n" if $verbose;
+
+        # Check that the desired regex was found in the output
+        my $match_count = grep {$_ =~ $match_rx} @cmd_output;
+        print "Expected $match_rx match count: $expect_match_count, got $match_count\n" if $verbose;
+        ok($match_count == $expect_match_count);
+	print "\n" if $verbose;
+    }
+
+    before each => sub {
+        HTFeed::Bunnies->new()->reset_queue;
+        get_dbh()->do("DELETE FROM feed_queue");
+        get_dbh()->do("DELETE FROM feed_zephir_items");
+    };
+
+    it "won't queue if missing bib data" => sub {
+        enqueue(
+            '-v -i -1 simple test test',
+            qr/Missing bibliographic data/,
+            1
+        );
+    };
+    it "can queue if there is bib data" => sub {
+	fake_bibdata(testvolume);
+        enqueue(
+            '-v -i -1 simple test test',
+            qr/simple test test: queued/,
+            1
+        );
+    };
+    it "won't queue an item already in the queue using -i" => sub {
+	fake_bibdata(testvolume);
+	my $same_command = '-v -i -1 simple test test';
+        enqueue(
+            $same_command,
+            qr/simple test test: queued/,
+            1
+        );
+        enqueue(
+            $same_command,
+            qr/simple test test: failure or skipped/,
+            1
+        );
+    };
+    it "needs -r or -R to queue an item already in the queue" => sub {
+	fake_bibdata(testvolume);
+        enqueue_testvolume(status => 'punted');
+
+        # Need to specify -r or -R for it to work
+        enqueue(
+            '-v -r -1 simple test test',
+            qr/simple test test: reset/,
+            1
+        );
+        enqueue(
+            '-v -R -1 simple test test',
+            qr/simple test test: reset/,
+            1
+        );
+    };
+    it "uses -u to reuse punted items rather than re-download" => sub {
+	fake_bibdata(testvolume);
+	enqueue_testvolume(status => 'punted');
+
+	# -u to reuse a punted item,
+	# only works in combination with a reset flag (-r/-R).
+	# (in this case i'm also re-routing stderr to devnull,
+	# because i'm doing such a bad thing that enqueue.pl
+	# is going to print its whole pod2usage to stderr, and I don't want it)
+	enqueue(
+            '-v -u -1 simple test test 2>/dev/null',
+            qr/empty result/,
```

Comment:
I'm unclear on what this actually tests. It seems to be looking for zero instances of "empty result" from STDOUT. Can you use an anchored regex to assert STDOUT is getting an empty string? Or perhaps look for the pod2usage string instead (instead of sending it to the bit bucket).

Reply:
When faced with "how do change this function to expect zero results" I weighed between `expect 1 case of /^$/` or `expect 0 cases of "empty result"`. Neither really satisfy.